### PR TITLE
`azurerm_firewall_policy_rule_collection_group`: add support for `http_headers`

### PR DESCRIPTION
**Pull request by wuxu92**
support `HTTPHeadersToInsert` in swagger.

### CODE REVIEW

File: internal/services/firewall/firewall_policy_rule_collection_group_resource_test.go [hunk 1]
```diff
@@ -794,6 +802,14 @@ resource "azurerm_firewall_policy_rule_collection_group" "test" {
...
       terminate_tls         = true
       web_categories        = ["News"]
+      http_headers {
+        name  = "head_foo"
+        value = "value_bar"
+      }
+      http_headers {
+        name  = "head_bar"
+        value = "value2"
+      }
```
Comment (stephybun) [inline internal/services/firewall/firewall_policy_rule_collection_group_resource_test.go:812]: It looks like we're supposed to be testing the update path for this property but these values are the same as the ones provided in the `completePremium` config. Should these be different?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @wuxu92.

One minor comment about the config values, otherwise this is looking good.

File: internal/services/firewall/firewall_policy_rule_collection_group_resource_test.go (hunk 1 above)
Comment (wuxu92) [reply internal/services/firewall/firewall_policy_rule_collection_group_resource_test.go]: Thanks, Updated.

```
--- PASS: TestAccFirewallPolicyRuleCollectionGroup_updatePremium (263.64s)
PASS
```

**Review by stephybun — APPROVED** [review]
Thanks @wuxu92 LGTM 👍

**Pull request merged**